src/Carousel.svelte

```diff
@@ -15,6 +16,14 @@
       (currentIndex =
         currentIndex != 0 ? currentIndex - 1 : loop ? items.length - 1 : 0),
   };
+  export let autoplayInterval = -1; // autoplay disactivated by default
```

Comment:
wondering if this can be made 'reactive'
as the code is now changing the interval doesn't affect the interval, this could be usefull for speeding up or down the autoplay function

Reply:
Not sure what you mean, autoplayInterval is exposed, so one can speed up / down the animation like so:

`<Carousel {items} let:payload {autoplayInterval}>`

Reply:
but it doesn't change does it ? once the interval is started, changing `autoplayInterval` doesn't do anything.

Reply:
ah... ok now I follow. I gotta go now, but I will have a look at it in the afternoon!

Reply:
Hey, haven't found time to look into that yet, but something else came up on an implementation I have for another project: what do you think the expected behaviour would be when, during autoplay, the user does click on the controls? I opted to stop autoplay in that case. Shall I reproduce this for this PR?

Reply:
I think this is becoming rather tricky.
I am wondering it play/pause should not be independent of the interval (like a second prop).
The expected behaviour would then be that it keeps playing, but at a different interval, this such that a consumer can also opt to pause when the interval changes by having the function that handles the interval change also stop the autoplay.

```js
/* in parent component /*
function handleIntervalChange() {
   autoplay = false
   interval = <newvalue>
}
```
I think this would also mean the animation stops **immediately** (staying on the current item).

The way to go seems (to me) to use a timeout instead of an interval, that way everytime the timeout is called it will use the, on that moment, current interval time. (this solves the reactivity of the interval) Also a timeout can be interrupted.